src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceProvider.cs

```diff
@@ -132,7 +132,8 @@ private void OnResolve(Type serviceType, IServiceScope scope)
 
         private void ValidateService(ServiceDescriptor descriptor)
         {
-            if (descriptor.ServiceType.IsGenericType && !descriptor.ServiceType.IsConstructedGenericType)
+            // Skip validation for generic service types
```

Comment:
We should add more description here explaining why we are skipping validation for generic service types.

Reply:
add a commit to address this